mlflow/data/delta_dataset_source.py

```diff
@@ -0,0 +1,114 @@
+
+from pyspark.sql import SparkSession, DataFrame
+
+from mlflow.data.dataset_source import DatasetSource
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+
+
+DeltaDatasetSourceType = TypeVar("DeltaDatasetSourceType", bound="DeltaDatasetSource")
+
+DATABRICKS_HIVE_METASTORE_NAME = "hive_metastore"
+# these two catalog names both points to the workspace local default HMS (hive metastore).
+DATABRICKS_LOCAL_METASTORE_NAMES = [DATABRICKS_HIVE_METASTORE_NAME, "spark_catalog"]
+# samples catalog is managed by databricks for hosting public dataset like NYC taxi dataset.
+# it is neither a UC nor local metastore catalog
+DATABRICKS_SAMPLES_CATALOG_NAME = "samples"
+
+
+class DeltaDatasetSource(DatasetSource):
+    def __init__(
+        self,
+        path: Optional[str] = None,
+        delta_table_name: Optional[str] = None,
+        delta_table_version: Optional[int] = None,
+    ):
+        if (path, delta_table_name).count(None) != 1:
+            raise MlflowException(
+                'Must specify exactly one of "path" or "table_name"',
+                INVALID_PARAMETER_VALUE,
+            )
+        self._path = path
+        self._delta_table_name = delta_table_name
+        self._delta_table_version = delta_table_version
+
+    @staticmethod
+    def _get_source_type() -> str:
+        return "delta_table"
+
+    def load(self, **kwargs) -> DataFrame:
+        """
+        Loads the dataset source as a Delta Dataset Source.
+        :return: An instance of `pyspark.sql.DataFrame`.
+        """
+        spark = (
+            SparkSession.builder.master("local[*]")
+            .config("spark.jars.packages", "io.delta:delta-core_2.12:2.2.0")
+            .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+            .config(
+                "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"
+            )
+            .getOrCreate()
+        )
+
+        spark_read_op = spark.read.format("delta")
+        if self._delta_table_version is not None:
+            spark_read_op = spark_read_op.option("versionAsOf", self._delta_table_version)
+
+        if self._path:
+            return spark_read_op.load(self._path)
+        else:
+            return spark_read_op.table(self._delta_table_name)
+
+    @property
+    def path(self) -> Optional[str]:
+        return self._path
+
+    @property
+    def delta_table_name(self) -> Optional[str]:
+        return self._delta_table_name
+
+    @property
+    def delta_table_version(self) -> Optional[int]:
+        return self._delta_table_version
+
+    @staticmethod
+    def _can_resolve(raw_source: Any):
+        return False
+
+    @classmethod
+    def _resolve(cls, raw_source: str) -> DeltaDatasetSourceType:
+        raise NotImplementedError
+
+    # check if table is in the Databricks Unity Catalog
+    def _is_databricks_uc_table(self):
```

Comment:
What happens if we aren't running in Databricks? Presumably that means that this isn't a UC table, right? Should we add an "are we running in Databricks?" check?